src/shared/selinux-util.c

```diff
@@ -436,7 +436,11 @@ int mac_selinux_get_create_label_from_exe(const char *exe, char **label) {
         if (sclass == 0)
                 return -ENOSYS;
 
-        return RET_NERRNO(security_compute_create_raw(mycon, fcon, sclass, label));
+        r = RET_NERRNO(security_compute_create_raw(mycon, fcon, sclass, label));
+        if (r < 0)
+                return mac_selinux_enforcing() ? r : -ENOSYS;
```

Comment:
as mentioned before, please handle this in service_determine_exec_selinux_label() instead.

note that the other user of mac_selinux_get_create_label_from_exe() wouldn't benefit of this ENOSYS mangling at all. hence, please in service_determine_exec_selinux_label() just handle this: log the error we get, and say we are ignoring it because of permissive mode.